MiniTeks/mks-server
# Add k8s deployment files and github automatic build workflows

## Added k8s deployment files and github automatic build workflows
*  Docker file  for mks-server image
*  Add k8s config files for db and mks-server
*  Github workflows  for automatic build
* Minor fixes

## What's going to make deployment happen
We start with creating our mks resources in the cluster, then we apply the k8s db config files to get the redis-db server up and running, and finally we apply the mks-server config files which in return creates a service account having cluster role and  permissions to allow it to talk to k8s API server.
Finally, we have github workflows to add the ability to auto build and push the latest image to github registry.

## How to test
0. Get the cluster configured, and create a namespace "*mks*"
```bash
oc new-project mks
```
1. Install tekton pipelines operator in the cluster.
```bash
oc apply --filename https://storage.googleapis.com/tekton-releases/pipeline/latest/release.yaml
```
2. Apply the files to create the mks resources inside the cluster
```bash
oc apply -f config/mksCRDs
```
3 Get db server up and running.
```bash
oc apply -f k8s/db-config
```
4. Get mks-server up and running.
```bash
oc apply -f k8s/mks-config
```
5. See the logs to verify
```bash
oc logs -f -n mks mks-server-<get the deployment name>
```
